dotCMS/src/main/java/com/dotcms/publisher/util/DependencyManager.java

```diff
@@ -1490,4 +1341,176 @@ Set getContainers() {
 	Set getFolders() {
 		return folders;
 	}
+
+	private class DependencyProcessorItem {
+		String assetKey;
+		AssetTypes assetTypes;
+
+		public DependencyProcessorItem(String assetKey, AssetTypes assetTypes) {
+			this.assetKey = assetKey;
+			this.assetTypes = assetTypes;
+		}
+	}
+
+	private class DependencyProcessor {
+		private BlockingQueue<DependencyProcessorItem> queue;
+		private Map<AssetTypes, Set<String>> assetsAlreadyProcessed;
+		private boolean finished;
+		private boolean started;
```

Comment:
these boolean should be mark as a AtomicBoolean to be thread safe

Reply:
done https://github.com/dotCMS/core/pull/19955/commits/803a1c5b4aae8f0a873736babf79c2ba79b734b3